### PR TITLE
Jb56659

### PR DESCRIPTION
**Pull request by LaakkonenJussi**

### CODE REVIEW

File: plugin/aml_plugin.c [hunk 1]
```diff
@@ -9,37 +9,132 @@
...
+static char **override_numbers = NULL;
+
+static void load_override_list()
+{
+	GKeyFile *keyfile;
+	GError *err = NULL;
+	gsize len;
+	int i;
+
+	keyfile = g_key_file_new();
```
Comment (monich) [inline plugin/aml_plugin.c:31]: Leak?

File: plugin/aml_plugin.c [hunk 2]
```diff
@@ -59,13 +154,17 @@ static int aml_plugin_init(void)
...
+	load_override_list();
 	return 0;
 }
 
 static void aml_plugin_exit(void)
 {
 	DBG("");
 	ofono_voicecall_filter_unregister(&aml_voicecall_filter);
+
+	if (override_numbers)
```
Comment (monich) [inline plugin/aml_plugin.c:166]: g_strfreev is NULL-tolerant

File: service/aml_battery_monitor.c [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+// TODO: when implementing battery monitor
+struct aml_battery_monitor {
+	guint8 battery_level;
+};
+
+struct aml_battery_monitor *aml_battery_monitor_new()
+{
+	struct aml_battery_monitor *bm;
+
+	bm = g_try_malloc0(sizeof(struct aml_battery_monitor));
```
Comment (monich) [inline service/aml_battery_monitor.c:16]: `g_new0(struct aml_battery_monitor,1)` (or `g_try_new0` which I think will never fail, just as `g_new0`) would provide better compile-time type checking

File: plugin/aml_plugin.c (hunk 1 above)
Comment (LaakkonenJussi) [reply plugin/aml_plugin.c:31]: Yep, thanks for pointing this out, cleaned the code a bit as well with ff5192ae8ab2c9bd797acf1ad9bed1846fa07fab

File: plugin/aml_plugin.c (hunk 2 above)
Comment (LaakkonenJussi) [reply plugin/aml_plugin.c]: I do seem to always think this as non-NULL-tolerant for some reason... I now recall doing this a lot ^^. Anyways, fixd with ff4886d27d3d29ff358675bbda2d0174c5e802e0

File: service/aml_battery_monitor.c (hunk 3 above)
Comment (LaakkonenJussi) [reply service/aml_battery_monitor.c:16]: Yeah, maybe those `g_try_new0()` could be preferred throughout the code. I'm a bit following ConnMan syntax here as it somehow feels the most natural for me. But changes made in 93edcda00bc16db09480221793883eeb9784e61b with some other stuff as well to cleanup code.

**Comment by LaakkonenJussi** [comment]
Closing this as #2 contains these and more, and is the complete stage 1 support.

**Pull request closed without merging**